lib/models/index.ts

```diff
@@ -0,0 +1,116 @@
+import { InjectedDependenciesParam, InjectedOptionsParam } from '../balena';
+
+/*
+Copyright 2016 Balena
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+const modelsTemplate = [
+	/**
+	 * @namespace application
+	 * @memberof balena.models
+	 */
+	'application',
+
+	/**
+	 * @namespace device
+	 * @memberof balena.models
+	 */
+	'device',
+
+	/**
+	 * @namespace key
+	 * @memberof balena.models
+	 */
+	'key',
+
+	/**
+	 * @namespace os
+	 * @memberof balena.models
+	 */
+	'os',
+
+	/**
+	 * @namespace config
+	 * @memberof balena.models
+	 */
+	'config',
+
+	/**
+	 * @namespace release
+	 * @memberof balena.models
+	 */
+	'release',
+];
+
+const tsModelsTemplate = [
+	/**
+	 * @namespace apiKey
+	 * @memberof balena.models
+	 */
+	'api-key',
+
+	/**
+	 * @namespace service
+	 * @memberof balena.models
+	 */
+	'service',
+
+	/**
+	 * @namespace image
+	 * @memberof balena.models
+	 */
+	'image',
+
+	/**
+	 * @namespace billing
+	 * @memberof balena.models
+	 * @description **Note!** The billing methods are available on Balena.io exclusively.
+	 */
+	'billing',
+];
+
+export = (deps: InjectedDependenciesParam, opts: InjectedOptionsParam) => {
+	const models = {};
+
+	modelsTemplate.forEach(moduleName => {
+		Object.defineProperty(models, moduleName, {
+			enumerable: true,
+			configurable: true,
+			get() {
+				const moduleFactory = require(`./${moduleName}`);
+				// We need the delete first as the current property is read-only
+				// and the delete removes that restriction
+				delete this[moduleName];
+				return (this[moduleName] = moduleFactory(deps, opts));
+			},
+		});
+	});
+
+	tsModelsTemplate.forEach(moduleName => {
+		Object.defineProperty(models, moduleName, {
+			enumerable: true,
+			configurable: true,
+			get() {
+				const moduleFactory = require(`./${moduleName}`).default;
+				// We need the delete first as the current property is read-only
+				// and the delete removes that restriction
+				delete this[moduleName];
+				return (this[moduleName] = moduleFactory(deps, opts));
+			},
+		});
+	});
```

Comment:
How about doing something like:
```
modelsTemplate =

	/**
	 * @namespace application
	 * @memberof balena.models
	 */
	application: () => require('./application')

	/**
	 * @namespace image
	 * @memberof balena.models
	 */
	image: () => require('./image').default
```
This way we don't need the second `tsModelsTemplate` code section and can even preserve the order (to avoid the unnecessary changes in the docs). Most importantly, it should work on browserify as well :)